Resources/ProxyProvider/DIRACCAProxyProvider.py

```diff
@@ -320,9 +327,11 @@ def __createCertM2Crypto(self):
     userCert.set_version(2)
     userCert.set_subject(self.__X509Name)
     userCert.set_serial_number(int(random.random() * 10 ** 10))
-    # Add extentionals
+    # Add extensions
```

Comment:
I very very strongly advice against using M2Crypto directly here. For two reasons:
* we encapsulated these libraries in the `X509*` classes as to avoid complication and the spread in it
* with this, the user can no longer rollback to pyGSI in case of issue
* if a user was to run this, with pyGSI for the rest of the system, this will lead in crashes because pyGSI and M2Crypto pollute each others

Reply:
X509 classes are not exposing the necessary functionality for the DIRACCAProxyProvider. This is not a "user-level" class and is only supposed to be used in a server environment where we can control the environment. In any case, if you suspect potential conflicts here (which did not show up in the tests yet), what would you suggest to do then ?

Reply:
then anyone using that should just make sure to always have `DIRAC_USE_M2CRYPTO` to `True`. Anyway, hopefully falling back to pyGSI will not prove necessary

Reply:
out of curiosity, what is exactly the aim of all that ? What are the features missing in the `X509*` classes ?

Reply:
I have created simple `Test_DIRACCAProxyProvider` tests that pass for both pyGSI and M2Crypto(with `DIRAC_USE_M2CRYPTO` `yes`/`no`), as I see. Could you explain what you mean by pyGSI and M2Crypto polluting each other, how to test it?

I also looked at the code and actually there used the X509* classes methods:
`
loadChainFromFile()
getCredentials()
loadChainFromString()
loadKeyFromString()
generateProxyToString()
generateProxyToString()
`
As for the purpose of this class, I will try to change the description to a clearer one. In short, then the goal is the proxy generation using local (DIRAC) CA credentials for users without any certificate.

I didn't find the implementation in X509* classes, for the part of the code where the certificate is created for the user, although I agree that it is better to use existing classes methods. So if you can tell me how to implement this piece of code using existing class methods, I'll be grateful.